Quarkus-quartz missing Class in native classpath

### Describe the bug

When using quarkus-quartz for scheduled tasks that must run non-concurrently, the application fails to deploy in native mode. The issue does not occur in JVM mode or during development (dev mode). The problem arises when attempting to deploy a native build with a configured Oracle datasource.

The following scheduled task was used in the application:
```java
@TimerCorrelationId
@Nonconcurrent
@Scheduled(executeWith = Scheduled.QUARTZ, cron = "0 */15 * * * ?", identity = "statuses")
public void statuses() {
    statusBean.processLastRun(false);
}
```
In native mode, the application fails at start with a connection error. After switching the configuration to use Quartz in RAM mode, the stack trace provided a better clue, indicating a missing class required for reflection : 
```
Caused by: org.quartz.JobPersistenceException: Couldn't remove trigger: io.quarkus.quartz.runtime.QuartzSchedulerImpl$NonconcurrentInvokerJob [See nested exception: java.lang.ClassNotFoundException: io.quarkus.quartz.runtime.QuartzSchedulerImpl$NonconcurrentInvokerJob]
  at org.quartz.impl.jdbcjobstore.JobStoreSupport.replaceTrigger(JobStoreSupport.java:1494)
  at org.quartz.impl.jdbcjobstore.JobStoreSupport$9.execute(JobStoreSupport.java:1464)
  at org.quartz.impl.jdbcjobstore.JobStoreSupport.executeInNonManagedTXLock(JobStoreSupport.java:3736)
  at org.quartz.impl.jdbcjobstore.JobStoreTX.executeInLock(JobStoreTX.java:95)
  at org.quartz.impl.jdbcjobstore.JobStoreSupport.replaceTrigger(JobStoreSupport.java:1460)
  at org.quartz.core.QuartzScheduler.rescheduleJob(QuartzScheduler.java:1107)
  at org.quartz.impl.StdScheduler.rescheduleJob(StdScheduler.java:322)
  at io.quarkus.quartz.runtime.QuartzSchedulerImpl.<init>(QuartzSchedulerImpl.java:240)
  ... 25 more
Caused by: java.lang.ClassNotFoundException: io.quarkus.quartz.runtime.QuartzSchedulerImpl$NonconcurrentInvokerJob
  at java.base@21.0.7/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:52)
  at java.base@21.0.7/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
  at java.base@21.0.7/java.lang.ClassLoader.loadClass(ClassLoader.java:121)
  at org.quartz.simpl.InitThreadContextClassLoadHelper.loadClass(InitThreadContextClassLoadHelper.java:73)
  at org.quartz.simpl.InitThreadContextClassLoadHelper.loadClass(InitThreadContextClassLoadHelper.java:79)
  at org.quartz.impl.jdbcjobstore.StdJDBCDelegate.selectJobForTrigger(StdJDBCDelegate.java:1643)
  at org.quartz.impl.jdbcjobstore.StdJDBCDelegate.selectJobForTrigger(StdJDBCDelegate.java:1609)
  at org.quartz.impl.jdbcjobstore.JobStoreSupport.replaceTrigger(JobStoreSupport.java:1475)

```
To resolve the issue, the missing class must be registered for reflection. Adding the following code resolves the problem and allows the native image to start successfully:

```java
@RegisterForReflection(classNames = "io.quarkus.quartz.runtime.QuartzSchedulerImpl$NonconcurrentInvokerJob")
public class FixNativeBuild {
}
```

### Expected behavior

The application should start successfully in native mode with quarkus-quartz and the Oracle datasource configured. Scheduled tasks should execute as expected without connection errors.

### Actual behavior

The application fails to start in native mode with a connection error. Switching to Quartz in RAM mode reveals a missing class required for reflection, which prevents the application from functioning correctly.

### How to Reproduce?

1. Create a Quarkus project with quarkus-quartz and configure a scheduled task using @Nonconcurrent and @Scheduled annotations.
2. Configure the application to use an Oracle datasource.
3. Build the application as a native image.
4. Start the native image and observe the connection error during startup.

### Output of `uname -a` or `ver`

MSYS_NT-10.0-22631 3.4.9-be826601.x86_64 2023-09-07 12:36 UTC x86_64 Msys

### Output of `java -version`

openjdk version "21.0.1" 2023-10-17 OpenJDK Runtime Environment (build 21.0.1+12-29) OpenJDK 64-Bit Server VM (build 21.0.1+12-29, mixed mode, sharing)

### Mandrel or GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

3.20.0

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

maven 3.9.9

### Additional information

I suggest to register for reflection NonconcurrentInvokerJob in Quarkus-quartz extension.

--- a/extensions/quartz/deployment/src/main/java/io/quarkus/quartz/deployment/QuartzProcessor.java
+++ b/extensions/quartz/deployment/src/main/java/io/quarkus/quartz/deployment/QuartzProcessor.java
@@ -263,7 +263,10 @@ public class QuartzProcessor {
         reflectiveClasses.addAll(getAdditionalConfigurationReflectiveClasses(config.triggerListeners(), TriggerListener.class));
         reflectiveClasses.addAll(getAdditionalConfigurationReflectiveClasses(config.jobListeners(), JobListener.class));
         reflectiveClasses.addAll(getAdditionalConfigurationReflectiveClasses(config.plugins(), SchedulerPlugin.class));
-
+        reflectiveClasses
+                .add(ReflectiveClassBuildItem.builder("io.quarkus.quartz.runtime.QuartzSchedulerImpl$NonconcurrentInvokerJob")
+                        .reason(getClass().getName())
+                        .methods().build());
         return reflectiveClasses;
     }
 

